### PR TITLE
TRC-41 - feat: Add `TrendingSuggestionList` component

### PR DESCRIPTION
**Pull request by mmcbride2929**
## TL/DR
This story involves creating a list component that returns a unordered list & renders the fetched data as list items.

## Overview of Changes

`index.tsx`
- Add React Query Provider around app

`TrendingSuggestionList.tsx`
- Create interface for items, and parameters
- Map data using our `TrendingItemSuggestion` component.

`TrendingSuggestionList.scss`
- Style component for default and loading state

`TrendingSuggestionList.stories.tsx`
- Create Story for Default and Loading State Instances

`TrendingSuggestionList.utils`
- Created dummy data in order to fetch an array of objects to map over.

`TrendingItemSuggestion.scss`
- Changed loading state color & margin between each list item

`TrendingSuggestionList.test.tsx`
- Create tests to make sure component is working as expected

### CODE REVIEW

**Comment by mmcbride2929** [comment]
The component is getting there, but still has a way to go. While working on the component I got stuck in a few places and had a couple of questions.

#### `TrendingSuggestionList.stories.tsx`
I want to create a story for my `TrendingSuggestionList` component in Storybook. The component uses an isLoading value that comes from fetching data with React Query instead of being passed in as a prop. How can I manipulate isLoading within the story if `isLoading` isn't passed within args?

#### `TrendingSuggestionList.test.tsx`
I'm writing a test to check that our component renders the correct number of <li> elements based on the data array length that we fetch. I'm trying to use find/query/getAllByTestId methods to grab the <li> elements and count them, but the test isn't populating the results the way I was hoping it would. 

I've tried using `await waitFor` to make sure the component renders after the data is fetched, but the <li> elements still won't populate the `suggestionListItems` variable. Specifically, the line `const SuggestionListItems = await screen.findAllByTestId('list-item')` isn't working as expected. Any ideas on what might be causing this issue?


#### `TrendingSuggestionList.tsx`
One thing I noticed was that I had to create an interface to type the single item being mapped in the `TrendingSuggestionList` component and we are also creating an interface for `metaData` within the `TrendingItemSuggestion` component. Is this usual or is there a way to avoid the repetition?

**Comment by wijohnst** [comment]
For your question re: the `isLoading` prop, this is a good example of separation of concerns and and also an example of "not fighting the framework", in this case Storybook. 

The data fetching should happen in a top-level component, something like `<TrendingSuggestions />`, which returns a `<TrendingSuggestionsList />` component. All business logic, in this case data fetching, should happen in this top-level component and pass and rendering data down to its children components.

File: apps/web/public/Data/SuggestedItems.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      "subRedditName": "r/SubbreditLink"
+    }
+  },
+  {
+    "id": "002",
+    "URL": "https://images.unsplash.com/photo-1515879218367-8466d910aaa4?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTB8fGNvZGV8ZW58MHx8MHx8&w=1000&q=80",
+    "trendingSuggestionHeading": "Search Preview Item Heading 2",
+    "trendingSuggestionSummary": "Search Preview Item Summary 2",
+    "targetSubreddit": {
+      "subRedditIcon": "<Icon><ReactIcon /></Icon>",
```
Comment (wijohnst) [inline apps/web/public/Data/SuggestedItems.json:18]: Looks like you are trying to pass React components via JSON. Does this work?

File: apps/web/src/components/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -52,7 +52,6 @@ const SearchBar = ({
   const handleOnChange = (): void => {
     const searchString = ref.current?.value;
     if (!!searchString) {
-      // commented bc func causes error onChange as it is not being passed from parent
       //  filterResults(searchString);
```
Comment (wijohnst) [inline apps/web/src/components/SearchBar/SearchBar.tsx:55]: Can you make a story to address this issue?

File: apps/web/src/components/TrendingItemSuggestion/TrendingItemSuggestion.scss [hunk 3]
```diff
@@ -5,18 +5,18 @@ li {
   max-height: 8.75rem;
   border-radius: 0.25rem;
   padding: 1rem;
-  gap: 10;
-}
-
-.loading {
-  background-color: #d9dbdc;
+  margin-bottom: 3px;
```
Comment (wijohnst) [inline apps/web/src/components/TrendingItemSuggestion/TrendingItemSuggestion.scss:8]: Concert to a relative unit. Whenever you find yourself typing `px` it should trigger the thought "Can I do this with a relative unit?"

File: apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.scss [hunk 4]
```diff
@@ -0,0 +1,13 @@
+ul {
+  background-color: var(--white);
+  padding: 0;
+  margin: 0;
+}
+
+.loading-suggestion-list {
+  padding: 20px 10px 20px 10px;
```
Comment (wijohnst) [inline apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.scss:8]: Relative unit.

File: apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  customTestId?: string;
+}
+
+const TrendingSuggestionList = ({
+  customTestId = 'suggestion-list',
+}: Props): React.ReactElement => {
+  const { isLoading, data, isFetching, isFetched } = useQuery(
+    'suggested-items',
+    async () => {
+      const response = await fetch('../../Data/SuggestedItems.json');
```
Comment (wijohnst) [inline apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.tsx:34]: I like the way you did this, but we should switch to a method that is closer to the actual user experience, so something like Mock Service Worker. 

https://mswjs.io/

I'll make a story for adding MSW to the project.

File: apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.tsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { useQuery } from 'react-query';
+import TrendingItemSuggestion from '../TrendingItemSuggestion/TrendingItemSuggestion';
+import './TrendingSuggestionList.scss';
+
+interface Item {
```
Comment (wijohnst) [inline apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.tsx:6]: This name could be more semantic. `Item` doesn't really tell me anything.

File: apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.tsx (hunk 5 above)
Comment (wijohnst) [inline apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.tsx:34]: Additionally, instead of native `fetch` API I think we should switch to something more robust like Axios. That should help with the typing issue that you mentioned.

Comment (wijohnst) [reply apps/web/src/components/TrendingSuggestionList/TrendingSuggestionList.tsx:34]: Finally, the `async` function that actually fetches the data can be moved to a special API file, which will exist at the top-level, so in the same place as the component that will call `useQuery`, in this case a top-level `<TrendingSuggestions />` component. The Axios functions that fetch or post data would be written in a `TrendingSuggestions.api.ts` file, that way we can easily write tests against those functions.

**Review by wijohnst — COMMENTED** [review]
Great start.

**Review by wijohnst — APPROVED** [review]
This looks good. 

My only concern is the `metaData` including React components, but that is something we can sort out later.
